addons/mail/static/src/messaging/entity/activity/activity.js

```diff
@@ -0,0 +1,200 @@
+odoo.define('mail.messaging.entity.Activity', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entity.core');
+
+function ActivityFactory({ Entity }) {
+
+    class Activity extends Entity {
+
+        /**
+         * @override
+         */
+        delete() {
+            this.env.rpc({
+                model: 'mail.activity',
+                method: 'unlink',
+                args: [[this.id]],
+            });
+            super.delete();
+        }
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        async fetchAndUpdate() {
+            const data = await this.env.rpc({
+                model: 'mail.activity',
+                method: 'activity_format',
+                args: [this.id],
+            });
+            this.update(data);
+            if (this.chatter) {
+                this.chatter.refresh();
+            }
+        }
+
+        /**
+         * @param {Object} param0
+         * @param {mail.messaging.entity.Attachment[]} param0.attachments
+         * @param {string|boolean} [param0.feedback=false]
+         */
+        async markAsDone({ attachments, feedback = false }) {
+            const attachmentIds = attachments.map(attachment => attachment.id);
+            await this.env.rpc({
+                model: 'mail.activity',
+                method: 'action_feedback',
+                args: [[this.id]],
+                kwargs: {
+                    attachment_ids: attachmentIds,
+                    context: this.chatter.context,
+                    feedback,
+                },
+            });
+            this.chatter.refresh();
+            this.delete();
+        }
+
+        /**
+         * @param {Object} param0
+         * @param {string} param0.feedback
+         * @returns {Object}
+         */
+        async markAsDoneAndSchedule({ feedback }) {
+            const action = await this.env.rpc({
+                model: 'mail.activity',
+                method: 'action_feedback_schedule_next',
+                args: [[this.id]],
+                kwargs: { feedback },
+            });
+            this.chatter.refresh();
```

Comment:
Also from the other PR, I think `refresh` should be called in _update if that is appropriate (but to check, not sure what it does).